addons/point_of_sale/static/src/app/models/utils/recursive_serialization.js

```diff
@@ -185,6 +185,10 @@ const deepSerialization = (
             if (typeof record[fieldName] === "number") {
                 result[fieldName] = record[fieldName];
             }
+            result[fieldName] =
+                typeof record[fieldName] === "number"
+                    ? record[fieldName]
+                    : parseInt(record[fieldName].split("_")[1]);
```

Comment:
Why not removed if part? 

```suggestion
            let value = record[fieldName];
            if (typeof value === "string") {
                value = parseInt(value.split("_")[1]);
            }
            result[fieldName] = value !== undefined ? value : false; 
```
Issue was generated because missing handling ID as string (offline data) as SAAS-18.1 here https://github.com/odoo/odoo/blob/d4e40e4be4233d91db80717891cef73c4fdec06c/addons/point_of_sale/static/src/app/models/related_models.js#L281-L285
Also you could add above line reference in you commit/pr